packages/rest/src/controllers/didcomm/connections/ConnectionsController.ts

```diff
@@ -0,0 +1,137 @@
+    @Query('theirDid') theirDid?: Did,
+    @Query('theirLabel') theirLabel?: string,
+  ) {
+    const connections = await request.user.agent.connections.findAllByQuery({
+      alias,
+      did,
+      theirDid,
+      theirLabel,
+      state,
+      outOfBandId,
+    })
+
+    return connections.map(connectionRecordToApiModel)
+  }
+
+  /**
+   * Retrieve connection record by connection id
+   * @param connectionId Connection identifier
+   * @returns ConnectionRecord
+   */
+  @Example<DidCommConnectionsRecord>(connectionsRecordExample)
+  @Get('/:connectionId')
+  public async getConnectionById(@Request() request: RequestWithAgent, @Path('connectionId') connectionId: RecordId) {
+    const connection = await request.user.agent.connections.findById(connectionId)
+
+    if (!connection) {
+      this.setStatus(404)
+      return apiErrorResponse(`connection with connection id "${connectionId}" not found.`)
+    }
+
+    return connectionRecordToApiModel(connection)
+  }
+
+  /**
+   * Deletes a connection record from the connection repository.
+   *
+   * @param connectionId Connection identifier
+   */
+  @Delete('/:connectionId')
+  public async deleteConnection(@Request() request: RequestWithAgent, @Path('connectionId') connectionId: RecordId) {
+    try {
+      this.setStatus(204)
+      await request.user.agent.connections.deleteById(connectionId)
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        this.setStatus(404)
+        return apiErrorResponse(`connection with connection id "${connectionId}" not found.`)
+      }
+
+      this.setStatus(500)
+      return apiErrorResponse(error)
+    }
+  }
+
+  /**
+   * Accept a connection request as inviter by sending a connection response message
+   * for the connection with the specified connection id.
+   *
+   * This is not needed when auto accepting of connection is enabled.
+   */
+  @Example<DidCommConnectionsRecord>(connectionsRecordExample)
+  @Post('/:connectionId/accept-request')
+  public async acceptRequest(@Request() request: RequestWithAgent, @Path('connectionId') connectionId: RecordId) {
+    try {
+      const connection = await request.user.agent.connections.acceptRequest(connectionId)
+      return connectionRecordToApiModel(connection)
+    } catch (error) {
+      if (error instanceof RecordNotFoundError) {
+        this.setStatus(404)
+        return apiErrorResponse(`connection with connection id "${connectionId}" not found.`)
+      }
+
+      this.setStatus(500)
+      return apiErrorResponse(error)
+    }
+  }
+
+  /**
+   * Accept a connection response as invitee by sending a trust ping message
+   * for the connection with the specified connection id.
+   *
+   * This is not needed when auto accepting of connection is enabled.
+   *
+   * @param connectionId Connection identifier
+   * @returns ConnectionRecord
+   */
+  @Example<DidCommConnectionsRecord>(connectionsRecordExample)
+  @Post('/:connectionId/accept-response')
```

Comment:
Is there a specific convention followed? Normally I go for `/accept-response/:connection-id` but I have no idea which is more common.

Reply:
this is restful practice. You do an operation on a connection. Otherwise you'd acces "AcceptResponse" resource with connection id

Format is always "<resource-category>/<resource-id>/<resource-specific-action>"

E.g. `books/12/delete` = From the books category, select the resource with id 12 and then perform the delete operation on that resource